packages/flutter_tools/test/general.shard/build_system/targets/asset_transformer_test.dart

```diff
@@ -0,0 +1,377 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:args/args.dart';
+import 'package:file/memory.dart';
+import 'package:file_testing/file_testing.dart';
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/build_system/tools/asset_transformer.dart';
+import 'package:flutter_tools/src/flutter_manifest.dart';
+
+import '../../../src/common.dart';
+import '../../../src/fake_process_manager.dart';
+
+void main() {
+  test('Invokes dart properly', () async {
+    final FileSystem fileSystem = MemoryFileSystem();
+    final BufferLogger logger = BufferLogger.test();
+    final Artifacts artifacts = Artifacts.test();
+
+    final File asset = fileSystem.file('asset.txt')..createSync()..writeAsStringSync('hello world');
+    const String outputPath = 'output.txt';
+
+    final FakeProcessManager processManager =
+        FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(
+        command: <Pattern>[
+          artifacts.getArtifactPath(Artifact.engineDartBinary),
+          'run',
+          'my_transformer',
+          RegExp('--input=.*'),
+          RegExp('--output=.*'),
+          '-f',
+          '--my_option',
+          'my_option_value',
+        ],
+        onRun: (List<String> args) {
+          final ArgResults parsedArgs = (ArgParser()
```

Comment:
can we also check the input path exists?

Reply:
I am not sure what the value of doing that here would be, could you explain?

FWIW, https://github.com/flutter/flutter/blob/2519e6ff31a8b06873dc7288cf7ccd80e2fe0a6f/packages/flutter_tools/lib/src/build_system/tools/asset_transformer.dart#L57 should throw if the file does not exist.

Reply:
I'm thinking that that's an implementation detail that could be refactored in the future

Reply:
Indeed. My original question still stands:

> I am not sure what the value of doing that here would be, could you explain?

Reply:
by checking that the input path exists, we're testing that in the future, someone doesn't refactor to the code, and we end up passing an input option argument to the wrong path, which does not actually have a file.

Reply:
See https://github.com/flutter/flutter/pull/143815#discussion_r1499903182

Reply:
ahh yeah, you're right, this LGTM